### PR TITLE
added create from path

### PR DESCRIPTION
**Pull request by alvarolorentedev**

### CODE REVIEW

File: enginefs.js [hunk 1]
```diff
@@ -201,6 +203,21 @@ router.all("/:infoHash/create", function(req, res) {
         res.end(JSON.stringify(getStatistics(engines[ih])));
     });
 });
+router.all("/create", function(req, res) {
+    var parsed
+    try {
+        var torrent = fs.readFileSync(req.body.from)
```
Comment (Ivshti) [inline enginefs.js]: why not async?

also, parse-torrent-file does take a file path, so you don't need `fs` at all. that's why i suggested it

**Review by Ivshti — CHANGES_REQUESTED** [review]

File: enginefs.js (hunk 1 above)
Comment (alvarolorentedev) [reply enginefs.js]: @Ivshti `parse-torrent-file` requires to read the file, it does not allow just use the path. maybe something changed from previous versions. 
the documentation shows a use very similar to what i have done.

https://github.com/feross/parse-torrent-file.

Regarding async no problem sounds like also a good option.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,57 @@
+Skip to content
+This repository
+Search
+Pull requests
+Issues
+Gist
+ @kanekotic
```
Comment (Ivshti) [inline .gitignore]: what's that mess? :)

File: enginefs.js [hunk 3]
```diff
@@ -201,6 +203,23 @@ router.all("/:infoHash/create", function(req, res) {
...
+
+router.all("/create", function(req, res) {
+    var torrent = fs.readFile(req.body.from,function (error, file) {
+        if (error){
+            res.writeHead(400, jsonHead);
+            res.end();
+            console.error(error)
+            return
+        } 
+        var parsed = parseTorrentFile(file)
```
Comment (Ivshti) [inline enginefs.js]: maybe this should be in a try-catch, but i'm not sure; check if the underlying implementation can throw

**Review by Ivshti — COMMENTED** [review]
check the comments, otherwise gtg

File: .gitignore (hunk 2 above)
Comment (kanekotic) [inline .gitignore]: 0_0

**Pull request merged**